styles/lib/carnival/features/_tables.scss

```diff
@@ -0,0 +1,367 @@
+      display: block;
+      position: relative;
+      top: -0.1rem;
+    }
+
+    .os-table:not(.os-unstyled-container) .os-caption-container {
+      display: table-caption;
+      caption-side: bottom;
+      prince-caption-page: all;
+      font-size: 0.8333333333rem;
+      line-height: 1.2;
+      > .os-title-label {
+        font-family: fonts.$mulish;
+        font-weight: bold;
+        color: colors.$R1;
+        text-transform: uppercase;
+      }
+      > .os-number {
+        font-family: fonts.$mulish;
+        font-weight: bold;
+        color: colors.$R1;
+      }
+      > .os-title {
+        font-family: fonts.$mulish;
+        font-weight: bold;
+      }
+      > .os-caption {
+        font-family: fonts.$plex;
+      }
+    }
+}
+
+@mixin column-header-table($book-table-color: colors.$black){
+  .os-table.os-column-header-container {
+    margin-bottom: 0.7rem;
+    display: table;
+    margin-left: auto;
+    margin-right: auto;
+    > table {
+      border-collapse: collapse;
+      > tbody
+        > tr {
+          > td:not(:only-of-type) {
+            font-family: fonts.$plex;
+            border-left: 0.025rem solid colors.$black;
+            border-bottom: 0.025rem solid colors.$black;
+            padding-top: 0.7rem;
+            padding-bottom: 0.7rem;
+            padding-right: 0.7rem;
+            padding-left: 0.7rem;
+          }
+          > td:not(:only-of-type):not(:last-child) {
+            border-left: none;
+            border-right: 0.1rem solid colors.$black;
+            font-family: fonts.$mulish;
+            font-weight: 700;
+          }
+        > tr:first-of-type {
+          border-top-color: $book-table-color;
+          border-top-width: 0.1rem;
+          border-top-style: solid;
+        }
+        > tr:last-of-type > td {
+          border-bottom-color: $book-table-color;
+          border-bottom-width: 0.1rem;
+          border-bottom-style: solid;
+        }
+        }
+    }
+  }
+}
+
+@mixin vertically-tight-table(){
+  .os-table.os-vertically-tight-container > table.vertically-tight {
+    > thead > tr > th {
+      padding-top: 0.16rem;
+      padding-bottom: 0.16rem;
+    }
+    > tbody > tr > td:not(.regular-spacing):not(.pseudo-class-for-precedence) {
+      padding-top: 0.16rem;
+      padding-bottom: 0.16rem;
+    }
+  }
+}
+
+@mixin table-from-exercise(){
+  .os-eoc [data-type=problem] > div .os-table > table {
+    max-width: 100%;
+    table-layout: fixed;
+    > tbody > tr > td:not(:only-of-type) {
+      text-align: center;
+      padding-top: 0.7rem;
+      padding-bottom: 0.7rem;
+      padding-right: 4px;
+      padding-left: 4px;
+    }
+  }
+  .os-eoc [data-type=exercise-question] > div .os-table > table {
+    max-width: 100%;
+    table-layout: fixed;
+    > tbody > tr > td:not(:only-of-type) {
+      text-align: center;
+      padding-top: 0.7rem;
+      padding-bottom: 0.7rem;
+      padding-right: 4px;
+      padding-left: 4px;
+    }
+  }
```

Comment:
since the properties are otherwise identical, this can become:

```scss
.os-eoc {
  [data-type=problem], [data-type=exercise-question] {
    > div .os-table > table {
      // specs
      > tbody > tr > td:not(:only-of-type) {
        // specs
      }
    }
  }
}
```